strawberryfields/tdm/tdmprogram.py

```diff
@@ -413,27 +413,34 @@ def context(self, *args, shift="default"):
     # pylint: disable=too-many-branches
     def compile(self, *, device=None, compiler=None):
         """Compile the time-domain program given a Strawberry Fields photonic hardware device specification.
-        
+
         Currently, the compilation is simply a check that the program matches the device.
 
         Args:
             device (~strawberryfields.api.DeviceSpec): device specification object to use for
                 program compilation
             compiler (str, ~strawberryfields.compilers.Compiler): Compiler name or compile strategy
                 to use. If a device is specified, this overrides the compile strategy specified by
-                the hardware :class:`~.DeviceSpec`.
+                the hardware :class:`~.DeviceSpec`. If no compiler is passed, the default "TD2"
+                compiler is used. Currently, the only other allowed compiler is "gaussian".
 
         Returns:
             Program: compiled program
         """
         if compiler == "gaussian":
             return super().compile(device=device, compiler=compiler)
 
-        if compiler == "TD2" and device is not None:
+        if device is not None:
+            device_layout = bb.loads(device.layout)
+
+            if device_layout.programtype["name"] != "tdm":
+                raise TypeError(
+                    'TDM compiler only supports "tdm" type device specification layouts. '
```

Comment:
```suggestion
                    "TDM compiler only supports "tdm" type device specifications. "
```
(perhaps needs to be Blacked?)

Reply:
Nope, there's a `"tdm"` inside of the string, so this is correct. Although it might be better to just write `tdm` instead. :thinking: